pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -429,21 +439,53 @@ func vagueUnder(actual, estimated time.Duration) string {
 
 var controlPlaneStatusTemplate = template.Must(
 	template.New("controlPlaneStatus").
-		Funcs(template.FuncMap{"shortDuration": shortDuration, "vagueUnder": vagueUnder}).
+		Funcs(template.FuncMap{"shortDuration": shortDuration, "vagueUnder": vagueUnder,
+			"updatingLine": func(elems []UpdatingClusterOperator) string {
+				if len(elems) == 0 {
+					return ""
+				}
+				var messages []string
+				for _, e := range elems {
+					message := e.Name
+					if e.Condition.Reason != "" {
+						message = fmt.Sprintf("%s: %s", e.Name, e.Condition.Reason)
+					}
+					messages = append(messages, message)
+				}
+				return fmt.Sprintf("Updating:        %s\n", strings.Join(messages, ", "))
+			},
+		}).
 		Parse(controlPlaneStatusTemplateRaw))
 
-func (d *controlPlaneStatusDisplayData) Write(f io.Writer) error {
+func (d *controlPlaneStatusDisplayData) Write(f io.Writer, detailed bool) error {
 	if d.Operators.Updated == d.Operators.Total {
 		_, err := f.Write([]byte(fmt.Sprintf("= Control Plane =\nUpdate to %s successfully completed at %s (duration: %s)\n", d.TargetVersion.target, d.CompletionAt.UTC().Format(time.RFC3339), shortDuration(d.Duration))))
 		return err
 	}
-	return controlPlaneStatusTemplate.Execute(f, d)
+	if err := controlPlaneStatusTemplate.Execute(f, d); err != nil {
+		return err
+	}
+	if detailed && len(d.Operators.UpdatingClusterOperators) > 0 {
+		table := tabwriter.NewWriter(f, 0, 0, 3, ' ', 0)
+		f.Write([]byte("\nUpdating Cluster Operators"))
+		_, _ = table.Write([]byte("\nNAME\tSINCE\tREASON\tMESSAGE\n"))
+		for _, o := range d.Operators.UpdatingClusterOperators {
+			_, _ = table.Write([]byte(o.Name + "\t"))
+			_, _ = table.Write([]byte(o.Condition.LastTransitionTime.UTC().Format(time.RFC3339) + "\t"))
+			_, _ = table.Write([]byte(o.Condition.Reason + "\t"))
+			_, _ = table.Write([]byte(o.Condition.Message + "\n"))
+		}
+		if err := table.Flush(); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 const controlPlaneStatusTemplateRaw = `= Control Plane =
 Assessment:      {{ .Assessment }}
 Target Version:  {{ .TargetVersion }}
-Completion:      {{ printf "%.0f" .Completion }}% ({{ .Operators.Updated }} operators updated, {{ .Operators.Updating }} updating, {{ .Operators.Waiting }} waiting)
+{{ updatingLine .Operators.UpdatingClusterOperators }}Completion:      {{ printf "%.0f" .Completion }}% ({{ .Operators.Updated }} operators updated, {{ .Operators.Updating }} updating, {{ .Operators.Waiting }} waiting)
```

Comment:
Let's keep the template readable by properly separating the line (then `updatingLine` should not return a line that ends with `\n` ofc)

Reply:
> "properly separating the line"

Because we want to hide the line when there is no CO that is updating.
That would leave an empty line, instead of hiding.

Reply:
That sound better to address with a conditional in the template though

Reply:
Let me try that.

Reply:
I still cannot avoid the line break (I called now I had tried a similar code before and hit the line break thing).
The relevant change is in the last commit https://github.com/openshift/oc/pull/1882/commits/1b8b4055d2b8b58b542e831ef84ca793db292554.
Let me know what I did wrong there.

Reply:
Check out https://pkg.go.dev/text/template#hdr-Text_and_spaces - it's possible to trim whitespace around the templating markup. I added a suggestion that shows how to do this.